### ISSUE TITLE
`no-unnecessary-polyfills`: always triggered on `core-js/{actual,full}/<feature>`

### ISSUE THREAD

**Issue by andreww2012** [sindresorhus/eslint-plugin-unicorn#2506]
Hello! When I import from `core-js/actual/<feature>` or `core-js/full/<feature>`, `no-unnecessary-polyfills` rule is incorrectly triggered, but if I import it from `es` or `stable`, it is not.

Example:

```ts
// Not reported:
import 'core-js/es/array/to-spliced';
import 'core-js/stable/array/to-spliced';

// Reported:
import 'core-js/actual/array/to-spliced';
import 'core-js/full/array/to-spliced';
```

Reproduction: https://stackblitz.com/edit/vitejs-vite-zrgjkh?file=src%2Fmain.js

```
pnpm test
```

**Comment by fregante** [comment]
> `no-unnecessary-polyfills` rule is incorrectly triggered

Why would that be incorrect?

```js
// Reported:
import 'core-js/actual/array/to-spliced';
```

If you include this in an environment that has `Array#toSpliced`, then it _is_ unnecessary.



> if I import it from `es` or `stable`, it is not

_This part_ sounds like the bug.
